### PR TITLE
[language builders] Get rid of base-builder-new

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Build all C/C++ projects in an image without rust, java,
atheris (python), or go.

### CODE REVIEW

**Review by inferno-chromium — APPROVED** [review]

**Review by oliverchang — APPROVED** [review]

File: infra/base-images/base-builder/Dockerfile [hunk 1]
```diff
@@ -191,14 +134,25 @@ RUN cd $SRC && \
     tar -xzv --strip-components=1 -f $SRC/oss-fuzz.tar.gz && \
     rm -rf examples $SRC/oss-fuzz.tar.gz
 
-COPY cargo compile compile_afl compile_dataflow compile_libfuzzer compile_honggfuzz \
-    compile_go_fuzzer precompile_honggfuzz precompile_afl debug_afl srcmap \
-    write_labels.py bazel_build_fuzz_tests /usr/local/bin/
-
-COPY detect_repo.py /opt/cifuzz/
-COPY ossfuzz_coverage_runner.go $GOPATH
+# Do precompiles before copying other scripts for better cache efficiency.
```
Comment (inferno-chromium) [inline infra/base-images/base-builder/Dockerfile:137]: COPY ossfuzz_coverage_runner.go $GOPATH
is important to keep

File: infra/build/functions/build_project.py [hunk 2]
```diff
@@ -257,11 +257,12 @@ def get_build_steps(  # pylint: disable=too-many-locals, too-many-statements, to
...
-                                              project.image,
-                                              project.fuzzing_language,
-                                              branch=config.branch,
-                                              test_images=config.test_images)
+  build_steps = build_lib.project_image_steps(
+      project.name,
+      project.image,
+      project.fuzzing_language,
+      branch=config.branch,
+      test_image_suffix=config.test_images)
```
Comment (inferno-chromium) [inline infra/build/functions/build_project.py:265]: config.test_images needs to be fixed here, it needs to be test_image_suffix

**Review by inferno-chromium — APPROVED** [review]
Please rebase on top of https://github.com/google/oss-fuzz/pull/6367, you need the go stuff in new base-builder Dockerfile.

File: infra/base-images/base-builder/Dockerfile (hunk 1 above)
Comment (inferno-chromium) [reply infra/base-images/base-builder/Dockerfile:137]: See https://github.com/google/oss-fuzz/pull/6335#issuecomment-909521828 for context.

Comment (inferno-chromium) [reply infra/base-images/base-builder/Dockerfile:137]: IGNORE THIS, i am fixing this properly now. i will fix it in base-builder-go

File: infra/base-images/base-builder-swift/Dockerfile [hunk 3]
```diff
@@ -14,6 +14,6 @@
 #
 ################################################################################
 
-FROM gcr.io/oss-fuzz-base/base-builder:xenial
+FROM gcr.io/oss-fuzz-base/base-builder-new:xenial
```
Comment (inferno-chromium) [inline infra/base-images/base-builder-swift/Dockerfile]: Can you add a FIXME: that need to move to base-builder and 20.04. need to remember this too. or ask @catenacyber for help.

**Review by inferno-chromium — APPROVED** [review]

File: infra/base-images/base-builder-swift/Dockerfile (hunk 3 above)
Comment (catenacyber) [reply infra/base-images/base-builder-swift/Dockerfile]: I plan to do that as part of https://github.com/google/oss-fuzz/pull/6312 after this PR is merged and the dust settled

**Comment by catenacyber** [comment]
Should suricata, a C project with rust, usr base-builder-rust instead of simple base-builder ?

What about ecc-diff-fuzzer which uses C, Rust and Go ?

**Comment by jonathanmetzman** [comment]
I think the only project this seems to have broken is net-snmp

File: infra/test_build_change.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import logging
+import sys
+
+import build_and_push_test_images
+
+
+def main():
```
Comment (oliverchang) [inline infra/test_build_change.py:7]: @jonathanmetzman Was this intended to be committed? If so, would you mind fixing the lint issues in this file? 
I'm getting lint errors on this from another PR.

Comment (jonathanmetzman) [reply infra/test_build_change.py:7]: Didn't mean to commit this. Removed in #6380.

File: infra/base-images/base-builder-swift/Dockerfile (hunk 3 above)
Comment (catenacyber) [reply infra/base-images/base-builder-swift/Dockerfile]: cc @jonathanmetzman @inferno-chromium I got swift to work with new ubuntu in #6312
Tested with
```
python3 infra/helper.py build_image --no-pull base-builder
python3 infra/helper.py build_image --no-pull base-builder-swift
python3 infra/helper.py build_image --no-pull swift-protobuf
python3 infra/helper.py build_fuzzers swift-protobuf
```

**Pull request merged**